File: playbook-java/src/main/java/io/infinitic/playbook/java/asyncMethod/Stop.java
package io.infinitic.playbook.java.asyncMethod;

import io.infinitic.clients.InfiniticClient;
import io.infinitic.playbook.java.asyncMethod.workflows.AsyncMethodWorkflow;

public class Stop {
    public static void main(String[] args) {
        try(InfiniticClient client = InfiniticClient.fromConfigResource("/infinitic.yml")) {
            // cancelling the existing PromisesWorkflow instances with the "promises" tag
            AsyncMethodWorkflow instance = client.getWorkflowByTag(AsyncMethodWorkflow.class, Start.asyncMethodTag);
            client.cancel(instance);
        }
    }
}

File: playbook-kotlin/src/main/kotlin/io/infinitic/playbook/kotlin/asyncMethod/Stop.kt
package io.infinitic.playbook.kotlin.asyncMethod

import io.infinitic.clients.InfiniticClient
import io.infinitic.playbook.kotlin.asyncMethod.workflows.AsyncMethodWorkflow

fun main() {
    InfiniticClient.fromConfigResource("/infinitic.yml").use { client ->
        // cancelling the existing PromisesWorkflow instances with the "promises" tag
        val instance = client.getWorkflowByTag(AsyncMethodWorkflow::class.java, tag)
        client.cancel(instance)
    }
}
